torch/testing/_internal/common_methods_invocations.py

```diff
@@ -4245,19 +4285,43 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
                    ref=np.i0,
                    decorators=(precisionOverride({torch.bfloat16: 3e-1,
                                                   torch.float16: 5e-1}),),
+                   skips=(
+                       SkipInfo('TestOpInfo', 'test_supported_backward',
+                                dtypes=(torch.float16, torch.bfloat16)),
+                   ),
                    dtypes=floating_types_and(torch.bfloat16),
                    dtypesIfCPU=floating_types_and(torch.bfloat16),
-                   dtypesIfCUDA=floating_types_and(torch.half, torch.bfloat16),
-                   supports_autograd=False),
+                   dtypesIfCUDA=floating_types_and(torch.half, torch.bfloat16)),
     UnaryUfuncInfo('special.i0e',
                    aten_name='special_i0e',
                    ref=scipy.special.i0e if TEST_SCIPY else _NOTHING,
                    decorators=(precisionOverride({torch.bfloat16: 3e-1,
                                                   torch.float16: 3e-1}),),
+                   skips=(
+                       SkipInfo('TestOpInfo', 'test_supported_backward',
+                                dtypes=(torch.float16, torch.bfloat16)),
+                   ),
                    dtypes=all_types_and(torch.bool, torch.bfloat16),
                    dtypesIfCPU=all_types_and(torch.bool, torch.bfloat16),
                    dtypesIfCUDA=all_types_and(torch.bool, torch.half, torch.bfloat16),
-                   supports_autograd=False,
+                   sample_inputs_func=sample_inputs_i0_i1,
+                   safe_casts_outputs=True),
+    UnaryUfuncInfo('special.i1',
+                   aten_name='special_i1',
+                   ref=np_unary_ufunc_integer_promotion_wrapper_with_astype(scipy.special.i1),
+                   decorators=(precisionOverride({torch.float: 1e-4}),),
+                   dtypes=all_types_and(torch.bool),
+                   dtypesIfCPU=all_types_and(torch.bool),
```

Comment:
For types, earlier it looked like there was vectorized support for bfloat16 and half, and on CUDA there's accommodation to promote to the accumulate type in the kernel, but `all_types` does not include half or bfloat16 -- should the half and bfloat16 code just be removed (for now)? Alternatively support for it should be added and tested

Reply:
Killed the support for `bfloat` and `half`. I think it would be better to add support for them when requested. Otherwise it would be a bit messy, 
* Even with upcasting for computation, the result won't be very much correct. Refer https://github.com/pytorch/pytorch/pull/56352#discussion_r626111962
* `i1 and i1e` CPU won't support half or bfloat16
* Since `i0 and i0e` depends on `i1 and i1e` for backward, `backward_dtypesIfCUDA` for `i0 and i0e` will be `torch.half, torch.bfloat16, torch.float32, torch.float64` while for `CPU` it will be`torch.float32, torch.float64`